### ISSUE TITLE
Changes to the behaviour of search?

### ISSUE THREAD

**Issue by mattus** [spotify/web-api#194]
Hello,

It looks like on/around the 21st of March a change was made to the behaviour of the search API (/v1/search) around the handling of the market parameter in that it, if omitted, defaults to US.

The current behaviour (omitting the market parameter):
```
$ curl -s -X GET "https://api.spotify.com/v1/search?q=artist%3A%22Mike+Posner%22+title%3A%22I+Took+A+Pill+In+Ibiza%22&type=track&limit=50" -H "Accept: application/json" | jq '.tracks.items[].available_markets[]' | sort | uniq
"CA"
"MX"
"US"
```

Same search but specifying a market of GB:
```
$ curl -s -X GET "https://api.spotify.com/v1/search?q=artist%3A%22Mike+Posner%22+title%3A%22I+Took+A+Pill+In+Ibiza%22&type=track&limit=50&market=GB" -H "Accept: application/json" | jq '.tracks.items[].available_markets[]' | sort | uniq
"AD"
"AR"
"AT"
"AU"
"BE"
"BG"
"BO"
"BR"
"CH"
"CL"
"CO"
"CR"
"CY"
"CZ"
"DE"
"DK"
"DO"
"EC"
"EE"
"ES"
"FI"
"FR"
"GB"
"GR"
"GT"
"HK"
"HN"
"HU"
"ID"
"IE"
"IS"
"IT"
"LI"
"LT"
"LU"
"LV"
"MC"
"MT"
"MY"
"NI"
"NL"
"NO"
"NZ"
"PA"
"PE"
"PH"
"PL"
"PT"
"PY"
"SE"
"SG"
"SK"
"SV"
"TR"
"TW"
"UY"
```

Would it be possible to revert to the previous behaviour?
Or is there a better way of finding what counties a track is available in?

Cheers,

Matt

**Comment by thelinmichael** [comment]
Thanks for opening this issue @mattuas! We're investigating this at the moment, we'll let you know when we're making changes.

**Comment by kalleprymne** [comment]
@thelinmichael, since we are also relying on the Search API (and omitting 'market' parameter), I wonder if it's possible to get an update/estimate on this issue?

Best regards / Kalle

**Comment by colinwillis1** [comment]
@thelinmichael, please help. we are also having serious issues since the search default was changed. Is it possible to simple revert the API back to the original default? That would solve re-scripting issues/costs and make life pretty sweet. Thanks for your help and attention!

**Comment by colinwillis1** [comment]
@thelinmichael. BTW and FYI - we did adjust our script for "by country" search but found 74 instances of track URIs not associated with any country and therefore "missed" ... these 74 instances occured over the 150 tracks we were interested in

**Comment by brki** [comment]
This is affecting me too; I'd be happy to see it fixed.

**Comment by colinwillis1** [comment]
There are now also track IDs that are not associated with any countries at all making playlist info on a track inaccurate

**Comment by Westie** [comment]
This is related to #230 

I would say this is a critical issue, given that we are no longer able to search using the Echonest set of APIs for music information - for music that is not available on Spotify at all for listening yet is in Spotify/Echonest's database.

**Comment by brki** [comment]
Unfortunately the fix for #274 does not fix this issue 😞

**Comment by brki** [comment]
It would be great to see this fixed.

My use case:  I have a list of song and artist names that come from a radio station's playlist.  I want to find the best-matching track (not necessarily a perfect match), if any, for all available markets.

If I have to query many or all possible markets separately, I'll hit the API's rate limit threshold.

**Comment by kudos** [comment]
This is biting me as well. I'm trying to find tracks on Spotify for users from any geo, today I discovered that setting the market helped find some tracks. I'd be completely fine with a special parameter for querying all markets, I completely understand changing defaults is challenging.

In the meantime, what sort of rate limiting do you guys do? If I have to loop over all markets to get matches I'll do that for now.

**Comment by juliank** [comment]
Are there any updates on this issue (@thelinmichael or somebody else from Spotify)? The bug definitely still exists.

**Comment by Christilut** [comment]
This is still a problem.

I was wondering why my results were different from the normal Spotify results but omitting `market` just defaults to `US` which makes finding non-English songs kind of impossible.

**Comment by kudos** [comment]
I loop over all the major markets until I find it now and I haven't had the rate limit issues I was concerned about above.

**Comment by Christilut** [comment]
Yep exactly what I've just added, but still a problem with the smaller markets but this will have to do for now

**Comment by elibroftw** [comment]
No updates on this for over 2 years? What's going on? I guess I'll have to loop over all the markets too, thus putting more strain on Spotify's API.

**Comment by matthewjumpsoffbuildings** [comment]
this bug is making life really difficult
